src/mpid/ch4/src/ch4_progress.h

```diff
@@ -14,28 +14,16 @@
 #include "ch4_impl.h"
 
 #undef FUNCNAME
-#define FUNCNAME MPIDI_Progress_test
+#define FUNCNAME MPIDI_Progress_hooks
 #undef FCNAME
 #define FCNAME MPL_QUOTE(FUNCNAME)
-MPL_STATIC_INLINE_PREFIX int MPIDI_Progress_test(int flags)
+MPL_STATIC_INLINE_PREFIX int MPIDI_Progress_hooks(void)
 {
     int mpi_errno, made_progress, num_of_hooks, i;
     mpi_errno = MPI_SUCCESS;
 
-    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_PROGRESS_TEST);
-    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_PROGRESS_TEST);
-
-#ifdef HAVE_SIGNAL
-    if (MPIDI_CH4_Global.sigusr1_count > MPIDI_CH4_Global.my_sigusr1_count) {
-        MPIDI_CH4_Global.my_sigusr1_count = MPIDI_CH4_Global.sigusr1_count;
-        mpi_errno = MPIDI_check_for_failed_procs();
-        if (mpi_errno)
-            MPIR_ERR_POP(mpi_errno);
-    }
-#endif
-
     num_of_hooks = OPA_load_int(&MPIDI_CH4_Global.registered_progress_hooks);
-    if (num_of_hooks && (flags & MPIDI_PROGRESS_HOOKS)) {
+    if (num_of_hooks) {
```

Comment:
FWIW, if `num_of_hooks == 0` then the below loop will never run, even without this outer conditional.

Reply:
Oh yes, that's right. I'll remove the conditional.

Reply:
I have addressed this in the latest version.